frontend/src/concepts/pipelines/topology/usePipelineTaskTopology.ts

```diff
@@ -13,123 +19,184 @@ import {
   parseVolumeMounts,
   translateStatusForNode,
 } from './parseUtils';
-import { KubeFlowTaskTopology } from './pipelineTaskTypes';
+import { PipelineTask } from './pipelineTaskTypes';
+
+const EMPTY_STATE: PipelineNodeModelExpanded[] = [];
+
+const getNestedNodes = (
+  spec: PipelineSpecVariable,
+  items: Record<string, TaskKF>,
+  components: PipelineComponentsKF,
+  executors: PipelineExecutorsKF,
+  runDetails?: RunDetailsKF,
+): [nestedNodes: PipelineNodeModelExpanded[], children: string[]] => {
+  const nodes: PipelineNodeModelExpanded[] = [];
+  const children: string[] = [];
+
+  Object.entries(items).forEach(([name, details]) => {
+    const componentRef = details.componentRef.name;
+    const component = components[componentRef];
+    const status = parseRuntimeInfoFromRunDetails(name, runDetails);
```

Comment:
here you are not passing in the executions, you should be so that we get accurate node status. then you can use logic like this which was used in the parent func to this one

```tsx
const status = executions
        ? parseRuntimeInfoFromExecutions(taskId, executions)
        : parseRuntimeInfoFromRunDetails(taskId, runDetails);
```